src/LibVLCSharp/Shared/LibVLC.cs

```diff
@@ -231,13 +231,63 @@ public LibVLC(params string[] options)
             _gcHandle = GCHandle.Alloc(this);
         }
 
+        /// <summary>
+        /// Create and initialize a libvlc instance.
+        /// This functions accept a list of &quot;command line&quot; arguments similar to the
+        /// main(). These arguments affect the LibVLC instance default configuration.
+        /// LibVLC may create threads. Therefore, any thread-unsafe process
+        /// initialization must be performed before calling libvlc_new(). In particular
+        /// and where applicable:
+        /// <para>- setlocale() and textdomain(),</para>
+        /// <para>- setenv(), unsetenv() and putenv(),</para>
+        /// <para>- with the X11 display system, XInitThreads()</para>
+        /// (see also libvlc_media_player_set_xwindow()) and
+        /// <para>- on Microsoft Windows, SetErrorMode().</para>
+        /// <para>- sigprocmask() shall never be invoked; pthread_sigmask() can be used.</para>
+        /// On POSIX systems, the SIGCHLD signalmust notbe ignored, i.e. the
+        /// signal handler must set to SIG_DFL or a function pointer, not SIG_IGN.
+        /// Also while LibVLC is active, the wait() function shall not be called, and
+        /// any call to waitpid() shall use a strictly positive value for the first
+        /// parameter (i.e. the PID). Failure to follow those rules may lead to a
+        /// deadlock or a busy loop.
+        /// Also on POSIX systems, it is recommended that the SIGPIPE signal be blocked,
+        /// even if it is not, in principles, necessary, e.g.:
+        /// On Microsoft Windows Vista/2008, the process error mode
+        /// SEM_FAILCRITICALERRORS flagmustbe set before using LibVLC.
+        /// On later versions, that is optional and unnecessary.
+        /// Also on Microsoft Windows (Vista and any later version), setting the default
+        /// DLL directories to SYSTEM32 exclusively is strongly recommended for
+        /// security reasons:
+        /// Arguments are meant to be passed from the command line to LibVLC, just like
+        /// VLC media player does. The list of valid arguments depends on the LibVLC
+        /// version, the operating system and platform, and set of available LibVLC
+        /// plugins. Invalid or unsupported arguments will cause the function to fail
+        /// (i.e. return NULL). Also, some arguments may alter the behaviour or
+        /// otherwise interfere with other LibVLC functions.
+        /// There is absolutely no warranty or promise of forward, backward and
+        /// cross-platform compatibility with regards to libvlc_new() arguments.
+        /// We recommend that you do not use them, other than when debugging.
+        /// </summary>
+        /// <param name="enableDebugLogs">enable verbose debug logs</param>
+        /// <param name="options">list of arguments (should be NULL)</param>
+        public LibVLC(bool enableDebugLogs = true, params string[] options)
```

Comment:
why `true` as a default value? How does the compiler choose when calling `new LibVLC()` ? Could you make the bool option required here?
Could you make the other constructor call this one? `this(false, options)`

Reply:
> why true as a default value? 

Because the idea is to have debug logging enabled by default so that users have clues immediately on what may be going wrong. And they would have to disable it explicitly for release mode (or not?). This is debatable... as mentioned in the first line of https://github.com/videolan/libvlcsharp/pull/165#issue-488120016

> How does the compiler choose when calling new LibVLC() ? How does the compiler choose when calling new LibVLC() ? Could you make the bool option required here? Could you make the other constructor call this one? this(false, options)

No need apparently, according to https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#overload-resolution

> If two candidates are judged to be equally good, preference goes to a candidate that doesn't have optional parameters for which arguments were omitted in the call. Overload resolution generally prefers candidates that have fewer parameters.

Reply:
So, it means that the second constructor is only called when a boolean is explicitely given, right?

We could then remove the = true with the same effect, right?

Reply:
Yeah, that's right. Will remove it.